source/io/module_javaScriptInvoke.js

```diff
@@ -400,17 +407,21 @@
             }
         };
 
-        var generateCompletionCallback = function (occurrencePointer, functionName, returnTypeName, returnValuePointer, errorStatusPointer, errorCodePointer, errorSourcePointer, completionCallbackStatus) {
+        var generateCompletionCallback = function (occurrencePointer, functionName, returnTypeName, returnValuePointer, errorStatusPointer, errorCodePointer, errorSourcePointer, completionCallbackStatus, isInternalFunction) {
             var completionCallback = function (returnValue) {
                 if (completionCallbackStatus.invocationState === completionCallbackInvocationEnum.FULFILLED) {
                     throw new Error('The completion callback was invoked more than once for ' + functionName + '.');
                 }
                 if (completionCallbackStatus.invocationState === completionCallbackInvocationEnum.REJECTED) {
                     throw new Error('The call to ' + functionName + ' threw an error, so this callback cannot be invoked.');
                 }
-                if (!(returnValue instanceof Error)) {
+                var errorStatus = Module.eggShell.dataReadBoolean(errorStatusPointer);
+                if (!(returnValue instanceof Error) && !errorStatus) {
                     tryUpdateReturnValue(functionName, returnTypeName, returnValuePointer, returnValue, errorStatusPointer, errorCodePointer, errorSourcePointer, completionCallbackStatus);
                 } else {
+                    if (isInternalFunction) {
```

Comment:
I don't think we want to throw in this case if errorStatus is true.

Reply:
But we also don't want to create a new errorMessage if errorStatus is true.

Reply:
For an internal function with error status true we should re-throw the exception.

internal functions should not throw exceptions, if they do they need to propagate up so they fail tests, etc.

This case is subtle, it may be worth adding a comment describing the case here